File: Graph.cs
using System;

// graph with adjacency matrix
public class Graph
{
    private readonly int[,] adjacency_matrix;
    private readonly int number_of_shops;
    public Graph(int[,] adjacency_matrix)
    {
        this.adjacency_matrix = adjacency_matrix;
        this.number_of_shops = adjacency_matrix.GetLength(0);
    }

    public int GetNumberOfShops()
    {
        return number_of_shops;
    }

    public int GetWeight(int first_vertex, int second_vertex)
    {
        return adjacency_matrix[first_vertex, second_vertex];
    }
}

File: src/graph.cc
//
// Created by geext on 7/8/18.
//

#include "graph.h"
namespace mtsp {
    Graph::Graph(const std::vector<std::vector<unsigned long>> &adjacency_matrix)
            : _adjacency_matrix(adjacency_matrix), _number_of_vertices(adjacency_matrix.size())
    {
    }

    std::size_t Graph::getNumberOfVertices() const
    {
        return _number_of_vertices;
    }

    unsigned long Graph::getWeight(std::size_t first_vertex, std::size_t second_vertex) const
    {
        return _adjacency_matrix[first_vertex][second_vertex];
    }
}